From 7a1f7806636ecaa6d320586dea5af2ed9f4ac4f0 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 16 Sep 2015 08:58:30 +0530
Subject: [PATCH] FOR_UPSTREAM [VPG]: drm/i915: Moves PPS delay to appropriate
 loc

Panel power delay is required for eDP displays to allow the panel to
suspend safely but is not required for external DP panels. This is fixed
by removing the delay for all displays and moved to a section where it
will affect only eDP display.

Change-Id: Ibec023d8db0d1343d735edb28c936e858273fd3a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5971
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 642659a..2b9f7ee 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1505,6 +1505,11 @@ void intel_edp_panel_off(struct intel_dp *intel_dp)
 	POSTING_READ(pp_ctrl_reg);
 
 	intel_dp->last_power_cycle = jiffies;
+
+	/* wait for power down delay */
+	msleep(intel_dp->panel_power_down_delay);
+
+	/* confirm panel exited from power down sequence */
 	wait_panel_off(intel_dp);
 
 	/* We got a reference when we enabled the VDD. */
@@ -4293,7 +4298,6 @@ intel_dp_link_down(struct intel_dp *intel_dp)
 	DP &= ~DP_AUDIO_OUTPUT_ENABLE;
 	I915_WRITE(intel_dp->output_reg, DP & ~DP_PORT_EN);
 	POSTING_READ(intel_dp->output_reg);
-	msleep(intel_dp->panel_power_down_delay);
 }
 
 static bool
-- 
1.9.1

